acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -331,6 +339,43 @@ impl<'a, B: BlackBoxFunctionSolver> ACVM<'a, B> {
             }
         }
     }
+
+    pub fn step_into_brillig_opcode(&mut self) -> StepResult<'a, B> {
```

Comment:
For comparison, heres some pseudocode (don't know if it compiles) for inspiration that uses early returns to avoid nesting:

```rust
pub fn step_into_brillig_opcode(&mut self) -> StepResult<'a, B> {
    let brillig = match &self.opcodes[self.instruction_pointer] {
        Opcode::Brillig(brillig) => brillig,
        _ => return StepResult::Status(self.solve_opcode()),
    };

    let witness = &mut self.witness_map;

    let should_skip = match BrilligSolver::<B>::should_skip(witness, brillig) {
        Ok(result) => result,
        Err(err) => return StepResult::Status(self.handle_opcode_resolution(Err(err))),
    };

    if should_skip {
        let resolution = BrilligSolver::<B>::zero_out_brillig_outputs(witness, brillig);
        return StepResult::Status(self.handle_opcode_resolution(resolution));
    }

    match BrilligSolver::new(witness, brillig, self.backend, self.instruction_pointer) {
        Ok(solver) => StepResult::IntoBrillig(solver),
        Err(..) => return StepResult::Status(self.handle_opcode_resolution(Err(()))),
    }
}
```

Reply:
Nice, thanks! Avoiding nesting with early returns is a muscle I haven't developed much yet. I'll blame years of Clojure programming on that :)